lib/hdb/hdb-ldap.c

```diff
@@ -1048,6 +1078,27 @@ LDAP_message2entry(krb5_context context, HDB * db, LDAPMessage * msg,
 #endif
     }
 
+    extensions = ldap_get_values_len(HDB2LDAP(db), msg, "krb5ExtendedAttributes");
+    if (extensions != NULL) {
+	size_t l;
+
+	ent->entry.extensions = malloc(sizeof(*(ent->entry.extensions)));
+	ent->entry.extensions->len = ldap_count_values_len(extensions);
+	ent->entry.extensions->val = (HDB_extension *) calloc(ent->entry.extensions->len, sizeof(HDB_extension));
+	if (ent->entry.extensions->val == NULL) {
+	    ret = ENOMEM;
+	    krb5_set_error_message(context, ret, "calloc: out of memory");
+	    goto out;
+	}
+	for (i = 0; i < ent->entry.extensions->len; i++) {
+	    decode_HDB_extension((unsigned char *) extensions[i]->bv_val,
```

Comment:
check return value, krb5_set_error_message with the same error code so we get a sane error string